adds "set injection for all" button to RUST + style tweaks

nl-208 opened the pull request:
:cl: Nl208
rscadd: Adds a "set injection rates for all" button to the RUST injector console's interface.
tweak: Adds some punctuation to the popup windows. Changed instructions to say "enter new injection rate between 1 and 100" instead of "0 and 100".
/:cl:
as per the last one, but with less leftover merge-commit crap. engineers rejoice.

I really should've gone to bed an hour ago.

SierraKomodo commented on code/modules/power/fusion/consoles/injector_control.dm:
```diff
@@ -18,6 +18,19 @@
+	if(href_list["global_rate"])
+		if(!lan || !fuel_injectors)
+			return TOPIC_NOACTION
+		var/new_injection_rate = input("Enter a new injection rate between 1 and 100. This will affect all injectors!", "Modifying injection rate") as num
```
Needs checks after this for the following after the input:
- The user did not move out of interaction range
- The user did not become incapacitated/unable to interact
- The user did not hit cancel (Null value result - Also update the input to be `as null|num` to allow properly cancelling the input)
- The user did not enter a value outside the range of 1 to 100

nl-208 replied:
quick note re: the out-of-bounds range bit. current behavior with the clamp call has it so that if you enter a value that's outside the accepted range, it automatically sets itself to either end of the range.

if you want, we can switch it to rejecting out-of-bounds input completely without too much grief, but considering SMESes' intake/output have the same behavior as what exists currently with the injection rate settings, I'd be in favor of keeping it as-is.

SierraKomodo replied:
Yeah, the clamp's fine to handle that case - I had written this comment before I'd gotten to the line that had the clamp.

SierraKomodo commented on code/modules/power/fusion/consoles/injector_control.dm:
```diff
@@ -18,6 +18,19 @@
+				return TOPIC_NOACTION
+			if(!new_injection_rate)
+				to_chat(user, SPAN_WARNING("That's not a valid injection rate."))
+				return TOPIC_NOACTION
```
Move this check to the recommendation above. Having it here will cause it to spam the user's chat with a message for every injector in the list.

SierraKomodo commented on code/modules/power/fusion/consoles/injector_control.dm:
```diff
@@ -18,6 +18,19 @@
+		var/new_injection_rate = input("Enter a new injection rate between 1 and 100. This will affect all injectors!", "Modifying injection rate") as num
+		for(var/obj/machinery/fusion_fuel_injector/F in fuel_injectors)
+			if(!istype(F))
+				return TOPIC_NOACTION
```
Istype checks here are redundant with `var/type/blah in list` loops.

```suggestion
```

SierraKomodo commented on code/modules/power/fusion/consoles/injector_control.dm:
```diff
@@ -18,6 +18,19 @@
+			if(!new_injection_rate)
+				to_chat(user, SPAN_WARNING("That's not a valid injection rate."))
+				return TOPIC_NOACTION
+			F.injection_rate = clamp(new_injection_rate, 0, 100) / 100
```
The changelog and input message indicate a limit of 1 to 100, but this clamp limits it from 0 to 100. Either update the messages or update this clamp.

SierraKomodo replied:
Also, this clamp and division operation should probably occur on `new_injection_rate` directly before the for loop, to avoid running the same math operations repeatedly.

nl-208 commented:
should have time to get all of the above finished by sometime tomorrow - see my note re: the out-of-bounds input sanity check, but the rest of it'll be pretty straightforward.

SierraKomodo commented on code/modules/power/fusion/consoles/injector_control.dm:
```diff
@@ -11,14 +11,27 @@
+			return TOPIC_NOACTION
+		if(!CanInteract(user,state))
+			return TOPIC_NOACTION
+		for(var/obj/machinery/fusion_fuel_injector/F in fuel_injectors)
```
Github is throwing a fit again. Let's try this again.

It should be safe to use `as anything` here to bypass the `istype()` checks this would otherwise call on every entry, since we know `fuel_injectors` will only ever contain the specific type we need.

```suggestion
		for(var/obj/machinery/fusion_fuel_injector/F as anything in fuel_injectors)
```

nl-208 replied:
done.

nl-208 committed:
adds "set injection for all" button to RUST + style tweaks


Sierra's changes, part 1

- adds a new variable for the clamped form of the user's input that's then applied to all injectors on the network. allows us to only do the math operation once.
- adds some sanity-checking for cancellation or out-of-bounds values, TODO be sure user hasn't walked away from the console/can still reach controls (i.e. isn't incapacitated or lying down)
- shifts the "invalid injection rate" case & message out of the for loop on the global function to prevent chat spam
Sierra's changes, part 2

- sanity checking for if the user's still able to interact with the console (hasn't walked away/gone prone)
- allows the input to properly accept null values (i.e. user cancellation)
Bypasses istype() calls on the for loop
